packages/components/src/components/radio-button/radio-button.ts

```diff
@@ -0,0 +1,234 @@
+            this.checked && !this.disabled
+              ? 'bg-primary border-primary text-white hover:bg-primary-500 hover:border-primary-500'
+              : this.disabled && !this.checked
+                ? 'border-neutral-500 text-neutral-500 hover:cursor-not-allowed'
+                : this.disabled && this.checked
+                  ? 'bg-neutral-500 text-white hover:cursor-not-allowed'
+                  : 'bg-transparent text-primary border-primary hover:bg-primary-100 hover:border-primary-500 hover:text-primary-500 cursor-pointer',
+            hasDefaultSlot && 'px-4',
+            this.hasFocus && 'focused-class',
+            hasDefaultSlot && 'button--has-label',
+            hasIconSlot && 'button--has-icon flex gap-2',
+            buttonSizeClass
+          )}"
+          aria-disabled=${this.disabled}
+          type="button"
+          value=${ifDefined(this.value)}
+          tabindex="${this.checked ? '0' : '-1'}"
+          @blur=${this.handleBlur}
+          @focus=${this.handleFocus}
+          @click=${this.handleClick}
+        >
+          <slot
+            name="icon"
+            part="icon"
+            class="${cx(
+              hasIconSlot && 'inline-flex relative items-center',
+              {
+                sm: 'text-base',
+                md: 'text-lg',
+                lg: 'text-xl'
+              }[this.size]
+            )}"
+            ?hidden=${!hasIconSlot}
+          ></slot>
+          ${hasDefaultSlot
+            ? html`<slot part="label" class="button__label inline-flex relative items-center whitespace-nowrap"></slot>`
+            : null}
+        </button>
+      </div>
+    `;
+  }
+  static styles = [
+    SolidElement.styles,
+    componentStyles,
+    css`
+      :host {
+        display: block;
+        width: min-content;
+      }
+
+      /* We use a hidden input so constraint validation errors work, since they don't appear to show when used with buttons. We can't actually hide it, though, otherwise the messages will be suppressed by the browser. */
+      .hidden-input {
+        all: unset;
+        position: absolute;
+        top: 0;
+        left: 0;
+        bottom: 0;
+        right: 0;
+        outline: dotted 1px red;
+        opacity: 0;
+        z-index: -1;
+      }
+
+      .lg-label {
+        height: 48px;
+      }
+
+      .lg-no-label {
+        height: 48px;
+        width: 48px;
+      }
+
+      .md-label {
+        height: 40px;
+      }
+
+      .md-no-label {
+        height: 40px;
+        width: 40px;
+      }
+
+      .sm-label {
+        height: 32px;
+      }
+
+      .sm-no-label {
+        height: 32px;
+        width: 32px;
+      }
+
+      :host(.sd-button-group__button--first:not(.sd-button-group__button--last)) button {
+        border-start-end-radius: 0;
+        border-end-end-radius: 0;
+      }
+
+      :host(.sd-button-group__button--inner) button {
+        border-radius: 0;
+      }
+
```

Comment:
We can use Tailwind classes here now since Mario merged his PR yesterday. See https://github.com/solid-design-system/solid/pull/653